### PR TITLE
 topology2: Build the topology for MTL chromebook

### PR DESCRIPTION
**Pull request by RDharageswari**
Build the topology for the MTL chromebook with Headset on SSP 2 and speakers on SSP 0.

### CODE REVIEW

File: tools/topology/topology2/cavs-rt5682.conf [hunk 1]
```diff
@@ -87,34 +96,34 @@ IncludeByKey.DEEPBUFFER_FW_DMA_MS{
 #
 #SSP Index: 0, Direction: duplex
 Object.Dai {
-	SSP."0" {
+	SSP.1 {
```
Comment (plbossart) [inline tools/topology/topology2/cavs-rt5682.conf:99]: does this index need to change? I understood from @ranj063 's explanation that this ."0" was just a way to make the object unique. And if it does need to change, could it be a separate patch because it should have to bearing on the result and no relationship with the headset/amplifier changes.

File: tools/topology/topology2/cavs-rt5682.conf [hunk 2]
```diff
@@ -61,6 +61,15 @@ Define {
 	DEEP_BUFFER_PIPELINE_SRC	'mixin.15.1'
 	DEEP_BUFFER_PIPELINE_SINK	'mixout.2.1'
 	DEEP_BUFFER_PCM_NAME		'Deepbuffer Jack Out'
+	HEADSET_CODEC_NAME		'SSP0-Codec'
+	HEADSET_PLAYBACK_PCM_CAPS	'HS Playback'
+	HEADSET_CAPTURE_PCM_CAPS	'HS Capture'
```
Comment (plbossart) [inline tools/topology/topology2/cavs-rt5682.conf:66]: if this the PCM name or the stream name? I don't get what the 'CAPS' stand for or relate to.

File: tools/topology/topology2/cavs-rt5682.conf [hunk 3]
```diff
@@ -61,6 +61,15 @@ Define {
...
+	HEADSET_SSP_DAI_INDEX		0
+	HEADSET_HW_CONFIG_NAME		'HS HWCFG'
+	SPEAKER_CODEC_NAME		'SSP1-Codec'
+	SPEAKER_PLAYBACK_PCM_CAPS	'SPK Playback'
+	SPEAKER_SSP_DAI_INDEX		1
+	SPEAKER_HW_CONFIG_NAME		'SPK HWCFG'
```
Comment (plbossart) [inline tools/topology/topology2/cavs-rt5682.conf:72]: Can we use Headset and Speaker? Acronyms are not fun to read.

File: tools/topology/topology2/cavs-rt5682.conf [hunk 4]
```diff
@@ -259,7 +268,7 @@ Object.PCM {
 		Object.Base.fe_dai."Speakers" {}
 
 		Object.PCM.pcm_caps."playback" {
-			name	"SSP1 Playback"
+			name	"$SPEAKER_PLAYBACK_PCM_CAPS"
```
Comment (plbossart) [inline tools/topology/topology2/cavs-rt5682.conf:271]: here it''s definitively pcm caps but is this really required to conflate caps and stream name?

File: tools/topology/topology2/cavs-rt5682.conf (hunk 2 above)
Comment (ranj063) [reply tools/topology/topology2/cavs-rt5682.conf:66]: This is the PCM capabilities name but maybe we can rename it to HEADSET_CAPTURE_PCM_STREAM_NAME/HEADSET_PLAYBACK_PCM_STREAM_NAME

File: tools/topology/topology2/cavs-rt5682.conf (hunk 1 above)
Comment (ranj063) [reply tools/topology/topology2/cavs-rt5682.conf:99]: does not need to change

File: tools/topology/topology2/cavs-rt5682.conf (hunk 4 above)
Comment (ranj063) [reply tools/topology/topology2/cavs-rt5682.conf:271]: yes this is required to connect the PCM with the host component in the kernel.

File: tools/topology/topology2/cavs-rt5682.conf (hunk 2 above)
Comment (RDharageswari) [reply tools/topology/topology2/cavs-rt5682.conf:66]: Done

**Review by ranj063 — APPROVED** [review]

File: tools/topology/topology2/sof-ace-tplg/CMakeLists.txt [hunk 5]
```diff
@@ -22,6 +22,11 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-mtl-rt711-4ch.bin,DEEPBUFFER_FW_DMA_MS
 
 "cavs-rt5682\;sof-mtl-max98357a-rt5682\;PLATFORM=mtl,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,\
 PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-mtl-max98357a-rt5682.bin,DEEPBUFFER_FW_DMA_MS=10"
+
+"cavs-rt5682\;sof-mtl-max98357a-rt5682-ssp2-ssp0\;PLATFORM=mtl,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,\
+PDM1_MIC_B_ENABLE=1,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-mtl-max98357a-rt5682.bin,\
+DEEPBUFFER_FW_DMA_MS=10,HEADSET_SSP_DAI_INDEX=2,SPEAKER_SSP_DAI_INDEX=0,HEADSET_CODEC_NAME=SSP2-Codec,\
+SPEAKER_CODEC_NAME=SSP0-Codec"
```
Comment (plbossart) [inline tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:29]: the only nit-pick I would have is that the information is duplicated with HEADSET_SSP_DAI_INDEX=2 and HEADSET_CODEC_NAME=SSP2-Codec

If you need to change that's 2 variables instead of a single value.

This can be fixed as an improvement, it's good enough for now.

**Review by plbossart — APPROVED** [review]

File: tools/topology/topology2/sof-ace-tplg/CMakeLists.txt (hunk 5 above)
Comment (ranj063) [reply tools/topology/topology2/sof-ace-tplg/CMakeLists.txt:29]: @plbossart that feature is coming. @jsarha is working on a string concat feature that will allow us to do this.

File: tools/topology/topology2/cavs-rt5682.conf [hunk 6]
```diff
@@ -139,7 +148,7 @@ Object.Pipeline {
 		index	1
 
 		Object.Widget.copier.1 {
-			stream_name	'SSP0 Playback'
+			stream_name	'$HEADSET_PLAYBACK_PCM_STREAM_NAME'
```
Comment (jsarha) [inline tools/topology/topology2/cavs-rt5682.conf:151]: A minor nitpick: Wouldn't it be cleaner to have one commit to introduce the stream name variable and another to change the name?

**Review by jsarha — APPROVED** [review]
Finally a topology PR I can understand fully on the first read. Approved with a minor nitpick.
